rascal-vscode-extension/src/RascalLSPConnection.ts

```diff
@@ -0,0 +1,213 @@
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+import * as vscode from 'vscode';
+import * as net from 'net';
+import * as os from 'os';
+import * as path from 'path';
+
+import { integer, LanguageClient, LanguageClientOptions, ServerOptions, StreamInfo } from 'vscode-languageclient/node';
+import { getJavaExecutable } from './auto-jvm/JavaLookup';
+import { RascalFileSystemProvider } from './RascalFileSystemProviders';
+import { VSCodeUriResolverServer } from './VSCodeURIResolver';
+
+
+
+// deploymode = (deployMode || testDeployMode)
+// jarPath = rascalExtensionContext!.asAbsolutePath(path.join('.', 'assets', 'jars', j)))
+// push return onto subscription stack!
+export async function activateLanguageClient(
+    { language, title, jarPath, vfsServer, isParametricServer = false, deployMode = true, devPort = -1 } :
+        {language: string, title: string, jarPath: string, vfsServer: VSCodeUriResolverServer, isParametricServer: boolean, deployMode: boolean, devPort: integer} )
+    : Promise<LanguageClient> {
+    const serverOptions: ServerOptions = deployMode
+        ? await buildRascalServerOptions(jarPath, isParametricServer)
+        : () => connectToRascalLanguageServerSocket(devPort) // we assume a server is running in debug mode
+            .then((socket) => <StreamInfo> { writer: socket, reader: socket});
+
+    const clientOptions = <LanguageClientOptions>{
+        documentSelector: [{ scheme: '*', language: language }],
+    };
+
+    const client = new LanguageClient(language, title, serverOptions, clientOptions, !deployMode);
+    //rascalExtensionContext!.subscriptions.push(client);
+
+    await client.start();
```

Comment:
does this work as expected? I get warnings from the typescript checker.

Reply:
I get no warnings in my ide? (and yes, it works, if this one failed, nothing would work).

Reply:
I will update my json modules and see if it goes away...